egs/librispeech/ASR/tdnn_lstm_ctc/asr_datamodule.py

```diff
@@ -383,3 +383,212 @@ def test_clean_cuts(self) -> CutSet:
     def test_other_cuts(self) -> CutSet:
         logging.info("About to get test-other cuts")
         return load_manifest(self.args.manifest_dir / "cuts_test-other.json.gz")
+
+
+import math
+import random
+import numpy as np
+from typing import Optional, Dict
+
+import torch
+
+from lhotse import CutSet
+
+class SpecAugment(torch.nn.Module):
+    """
+    SpecAugment performs three augmentations:
+    - time warping of the feature matrix
+    - masking of ranges of features (frequency bands)
+    - masking of ranges of frames (time)
+
+    The current implementation works with batches, but processes each example separately
+    in a loop rather than simultaneously to achieve different augmentation parameters for
+    each example.
+    """
+
+    def __init__(
+        self,
+        time_warp_factor: Optional[int] = 80,
+        num_feature_masks: int = 1,
+        features_mask_size: int = 13,
+        num_frame_masks: int = 1,
+        frames_mask_size: int = 70,
+        max_frames_mask_fraction: float = 0.2,
```

Comment:
You can probably change the defaults here too, since the behavior/meaning of those options has changed.
And make sure to correctly document the new meaning of the options, e.g.:
clarify frames_mask_size is per masked region, num_frame_masks is the maximum (it's also constrained by max_frames_mask_fraction), and that max_frames_mask_fraction is the total masked fraction including all masked regions (the expected/mean fraction will be less than half of this, as we randomly select the mask sizes between 0 and the computed maximum mask lengths).